docs/internals/optimizer.rst

```diff
@@ -853,19 +876,29 @@ It tries to match patterns like ``X + 0`` on each subexpression.
 During the matching procedure, it resolves variables to their currently
 assigned expressions to be able to match more deeply nested patterns
 even when the code is in pseudo-SSA form.
+In fact, running CommonSubexpressionEliminator before helps this component track equivalent subexpressions.
 
 Some of the patterns like ``X - X -> 0`` can only be applied as long
 as the expression ``X`` is movable, because otherwise it would remove its potential side-effects.
 Since variable references are always movable, even if their current
 value might not be, the ExpressionSimplifier is again more powerful
-in split or pseudo-SSA form.
+in :ref:`expression-split <expression-splitter>` or :ref:`pseudo-SSA <ssa-transform>` form.
+
+Some of the simplifications destroy the :ref:`expression-split <expression-splitter>` form.
```

Comment:
As above I would phrase it as the expression split form not being guaranteed to be preserved